src/utils.py

```diff
@@ -68,13 +84,58 @@ def analyzeFileQ2(fileName):
                 unigramDict['<S>'] = unigramDict.get('<S>', 0) + 1
                 newLine = False
         else:
+            # counting <S> and <E> for current sentence
+            unigramCount += 2
             unigramDict['<E>'] = unigramDict.get('<E>', 0) + 1
```

Comment:
@romantz if line is empty shouldnt we skip it

Reply:
We do, but an empty line indicates the end of a sentence, so it's a good place to process <S> and <E> segments/tags